src/App/App.tsx

```diff
@@ -1,47 +1,23 @@
 import React, { Component, ReactElement } from 'react';
 import isNil from 'lodash/isNil';
+import 'normalize.css';
+import 'typeface-roboto/index.css';
 
 import storage from '../utils/storage';
 import { makeLogin, isTokenExpire } from '../utils/login';
 import Loading from '../components/Loading';
 import LoginModal from '../components/Login';
 import Header from '../components/Header';
 import { Container, Content } from '../components/Layout';
-import RouterApp from '../router';
 import API from '../utils/api';
-import 'typeface-roboto/index.css';
 import '../utils/styles/global';
-import 'normalize.css';
 import Footer from '../components/Footer';
-import { FormError } from '../components/Login/Login';
-import { PackageInterface } from '../components/Package/Package';
-
-interface AppContextData {
-  logoUrl: string;
-  scope: string;
-  isUserLoggedIn: boolean;
-  packages: PackageInterface[];
-  user: {
-    username?: string;
-  };
-}
-export const AppContext = React.createContext<AppContextData>({
-  logoUrl: window.VERDACCIO_LOGO,
-  user: {},
-  scope: window.VERDACCIO_SCOPE || '',
-  isUserLoggedIn: false,
-  packages: [],
-});
-const AppContextProvider = AppContext.Provider;
-export const AppContextConsumer = AppContext.Consumer;
-
-export interface AppStateInterface extends AppContextData {
-  error?: FormError;
-  showLoginModal: boolean;
-  isLoading: boolean;
-}
-export default class App extends Component<{}, AppStateInterface> {
-  public state: AppStateInterface = {
+
+import AppRoute from './AppRoute';
+import { AppProps, AppContextProvider } from './AppContext';
+
+export default class App extends Component<{}, AppProps> {
```

Comment:
The `AppProps` name is a little misleading here. It's used for the state of a class component but its name is `AppProps` :/ It's not immediately clear that this is on purpose.
I would suggest renaming the import
```typescript
import { AppProps as AppState, AppContextProvider } from './AppContext';
```

Reply:
Good point Tom, but as I am working on the refactor of the App Component, I believe we can merge like it is now 😉

Reply:
Ok @priscilawebdev , it's fine, but I will memorize this :P later on.